### ISSUE TITLE
Looks like we have another one like Issue #837

### ISSUE THREAD

**Issue by Easystep2** [microsoft/navcontainerhelper#1822]
Looks like we have another one like Issue  #837 here called **Entilement**

Used the latest BCContainerHelper version 2.07.

After adding the "Entitlement" into the list it throws an error:

BcContainerHelper version 2.0.7
```
Copying Al Source Files from C:\ProgramData\BcContainerHelper\Extensions\Original-18.0.23013.23532-NL-al\* to C:\Users\\*******\Documents\AL\BaseApp-BC180
WARNING: Unexpected error while handling AzureADApplicationAutomationBaseApp.Entitlement.al
Exception calling "Invoke" with "3" argument(s): "Cannot convert value ""Azure" to type "System.Int32". Error: "Input string was not in a correct format.""
At C:\Users\\*****\Documents\WindowsPowerShell\Modules\BcContainerHelper\2.0.7\AppHandling\Copy-AlSourceFiles.ps1:87 char:39
+ ...        $newFilename = "$($alFileStructure.Invoke($type, $id, $name))"
+                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : PSInvalidCastException


```
Using the script:

```
Clear-Host
Set-Location (Split-Path $MyInvocation.MyCommand.Path)
. '..\Tools\es2ContainerHelperFunctions.ps1'

Write-Host  -ForegroundColor green "Installing Latest version of the Module BCContainerHelper"
Get-LatestVersionBCContainerHelper

<# Set Business Central Instance and tenant data #>
$BCServerInstance = Read-Host -Prompt 'Input the name of the new instance'
$alProjectFolder = "C:\Users\\*****\Documents\AL\BaseApp-" + $BCServerInstance

Write-Host "Pulling Base App and create new Project"
$myModifiedObjects = @()
$alFileStructure = { Param ([string] $type, [int] $id, [string] $name) 
    if ($myModifiedObjects.Contains("$type$id")) {
        $folder = "Modified"
    }
    elseif (($id -ge 50000 -and $id -le 59999) -or ($id -gt 1999999999)) {
        $folder = "My Code"
    }
    else {
        $folder = "src"
    }

    switch ($type) {
        "enum"              { "$folder\Enum\$($name).Enum.al" }
        "page"              { "$folder\Page\$($name).Page.al" }
        "table"             { "$folder\Table\$($name).Table.al" }
        "codeunit"          { "$folder\Codeunit\$($name).Codeunit.al" }
        "report"            { "$folder\Report\$($name).Report.al" }
        "query"             { "$folder\Query\$($name).Query.al" }
        "xmlport"           { "$folder\XmlPort\$($name).xmlPort.al" }
        "profile"           { "$folder\Profile\$($name).Profile.al" }
        "dotnet"            { "$folder\Dotnet\$($name).Dotnet.al" }
        "enumextension"     { "$folder\Enumext\$($name).EnumExt.al" }
        "pageextension"     { "$folder\Pageext\$($name).PageExt.al" }
        "tableextension"    { "$folder\Tableext\$($name).TableExt.al" }
        "interface"         { "$folder\Interface\$($name).Interface.al" }
        "pagecustomization" { "$folder\Pageeext\$($name).PageCust.al" }
        "entitlement"       { "$folder\Entitlement\$($name).PageCust.al" }
        "controladdin"      { "$folder\Controladdin\$($name).al" }
        "requestpage"       { "$folder\Report\$($name).al" }  
        ".rdlc"             { "$folder\Report\$name$type" }
        ".docx"             { "$folder\Report\$name$type" }
        ".xlf"              { "$folder\Translations\$($name).xlf" }
        default { throw "Unknown type '$type'" }
    }
}

If ($BCServerInstance) {
    Create-AlProjectFolderFromBcContainer -containerName $BCserverInstance `
                                          -alProjectFolder $alProjectFolder  `
                                          -useBaseLine `
                                          -useBaseAppProperties  `
                                          -addGIT `
                                          -alFileStructure $alFileStructure

}
else {
        write-Host -ForegroundColor red "D365 BC Server Instance $BCServerInstance not found!"  
}
Pop-Location
```

**Comment by freddydk** [comment]
You can try to grab the Copy-AlSourceFiles from here: https://raw.githubusercontent.com/microsoft/navcontainerhelper/dev/AppHandling/Copy-AlSourceFiles.ps1 and see if that works.

**Comment by Easystep2** [comment]
Copying Al Source Files from C:\ProgramData\BcContainerHelper\Extensions\Original-18.0.23013.23532-NL-al\* to C:\Users\RickBlom\Documents\AL\BaseApp-BC180
WARNING: Unexpected error while handling BaseAppObjectsExec.PermissionSet.al
Exception calling "Invoke" with "3" argument(s): "Cannot convert value "*" to type "System.Int32". Error: "Input string was not in a correct format.""
At C:\Users\RickBlom\Documents\WindowsPowerShell\Modules\BcContainerHelper\2.0.7\AppHandling\Copy-AlSourceFiles.ps1:87 char:39
+ ...        $newFilename = "$($alFileStructure.Invoke($type, $id, $name))"
+                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], ParentContainsErrorRecordException
    + FullyQualifiedErrorId : PSInvalidCastException

**Comment by freddydk** [comment]
Apparently 18 also has permissionset and permissionsetextension.

**Comment by Easystep2** [comment]
Works like a charm using following script:

# ------------------------------------------------------------------------------------------------
# Copyright 2020 (c) EasyStep2 BV. All rights reserved. 
# This file is part of EasyStep2 Powershell Tools.
#
#  EasyStep2 Powershell Tools is free software: you can redistribute it and/or modify
#  it under the terms of the GNU General Public License as published by
#  the Free Software Foundation, either version 3 of the License, or
#  (at your option) any later version.
#
#  EasyStep2 software is distributed in the hope that it will be useful,
#  but WITHOUT ANY WARRANTY; without even the implied warranty of
#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#  GNU General Public License for more details.
#
#  You should have received a copy of the GNU General Public License
#  along with EasyStep2 software. If not, see <https://www.gnu.org/licenses/>.
# ------------------------------------------------------------------------------------------------
Clear-Host
Set-Location (Split-Path $MyInvocation.MyCommand.Path)
. '..\Tools\es2ContainerHelperFunctions.ps1'

Write-Host  -ForegroundColor green "Installing Latest version of the Module BCContainerHelper"
Get-LatestVersionBCContainerHelper

<# Set Business Central Instance and tenant data #>
$BCServerInstance = Read-Host -Prompt 'Input the name of the new instance'
$alProjectFolder = "C:\Users\RickBlom\Documents\AL\BaseApp-" + $BCServerInstance

Write-Host "Pulling Base App and create new Project"
$myModifiedObjects = @()
$alFileStructure = { Param ([string] $type, [int] $id, [string] $name) 
    if ($myModifiedObjects.Contains("$type$id")) {
        $folder = "Modified"
    }
    elseif (($id -ge 50000 -and $id -le 59999) -or ($id -gt 1999999999)) {
        $folder = "My Code"
    }
    else {
        $folder = "src"
    }

    switch ($type) {
        "enum"                   { "$folder\Enum\$($name).Enum.al" }
        "page"                   { "$folder\Page\$($name).Page.al" }
        "table"                  { "$folder\Table\$($name).Table.al" }
        "codeunit"               { "$folder\Codeunit\$($name).Codeunit.al" }
        "report"                 { "$folder\Report\$($name).Report.al" }
        "query"                  { "$folder\Query\$($name).Query.al" }
        "xmlport"                { "$folder\XmlPort\$($name).xmlPort.al" }
        "profile"                { "$folder\Profile\$($name).Profile.al" }
        "dotnet"                 { "$folder\Dotnet\$($name).Dotnet.al" }
        "enumextension"          { "$folder\Enumext\$($name).EnumExt.al" }
        "pageextension"          { "$folder\Pageext\$($name).PageExt.al" }
        "tableextension"         { "$folder\Tableext\$($name).TableExt.al" }
        "interface"              { "$folder\Interface\$($name).Interface.al" }
        "pagecustomization"      { "$folder\Pageeext\$($name).PageCust.al" }
        "requestpage"            { "$folder\Report\$($name).RequestPage.al" }
        "controladdin"           { "$folder\Controladdin\$($name).ControlAddin.al" }
        "permissionsetextension" { "$folder\Permisionsetext\$($name).PermissionsetExt.al" }
        "permissionset"          { "$folder\Permisions\\$($name).xml" }
        "entitlement"            { "$folder\Entitlements\$($name).json" }
        ".xlf"                   { "$folder\Translations\$($name).xlf" }
        ".rdlc"                  { "$folder\Report\$name$type" }
        ".docx"                  { "$folder\Report\$name$type" }
        default { throw "Unknown type '$type'" }
    }
}

If ($BCServerInstance) {
    Create-AlProjectFolderFromBcContainer -containerName $BCserverInstance `
                                          -alProjectFolder $alProjectFolder  `
                                          -useBaseLine `
                                          -useBaseAppProperties  `
                                          -addGIT `
                                          -alFileStructure $alFileStructure
}
else {
        write-Host -ForegroundColor red "D365 BC Server Instance $BCServerInstance not found!"  
}
Pop-Location

**Comment by freddydk** [comment]
Shipped in 2.0.9
